geps/gep-1713/index.md

```diff
@@ -0,0 +1,408 @@
+	//
+	// At least one Listener MUST be specified.
+	//
+	// Note: this is the same Listener type in the GatewaySpec struct
+    Listeners []Listener 
+}
+
+// ListenerSetStatus defines the observed state of ListenerSet.
+type ListenerSetStatus struct {
+    // Listeners provide status for each unique listener port defined in the Spec.
+	//
+	// +optional
+	// +listType=map
+	// +listMapKey=name
+	// +kubebuilder:validation:MaxItems=64
+	//
+	// Note: this is the same ListenerStatus type in the GatewayStatus struct
+	Listeners []ListenerStatus `json:"listeners,omitempty"`
+	
+	// Conditions describe the current conditions of the ListenerSet.
+	//
+	// Implementations should prefer to express ListenerSet conditions
+	// using the `GatewayConditionType` and `GatewayConditionReason`
+	// constants so that operators and tools can converge on a common
+	// vocabulary to describe Gateway state.
+	//
+	// Known condition types are:
+	//
+	// * "Accepted"
+	// * "Programmed"
+	//
+	// +optional
+	// +listType=map
+	// +listMapKey=type
+	// +kubebuilder:validation:MaxItems=8
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+// ParentGatewayReference identifies an API object including its namespace,
+// defaulting to Gateway.
+type ParentGatewayReference struct {
+	// Group is the group of the referent.
+	//
+	// +optional
+	// +kubebuilder:default="gateway.networking.k8s.io"
+	Group *Group `json:"group"`
+
+	// Kind is kind of the referent. For example "Gateway".
+	//
+	// +optional
+	// +kubebuilder:default=Gateway
+	Kind *Kind `json:"kind"`
+
+	// Name is the name of the referent.
+	Name ObjectName `json:"name"`
+
+	// Namespace is the namespace of the referenced object. When unspecified, the local
+	// namespace is inferred.
+	//
+	// Support: Core
+	//
+	// +optional
+	Namespace *Namespace `json:"namespace,omitempty"`
+}
+```
+
+#### YAML
+
+The following example shows a `Gateway` with an HTTP listener and two child HTTPS `ListenerSets` with unique hostnames and certificates. 
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  name: parent-gateway
+spec:
+  gatewayClassName: example
+  listeners:
+  - name: foo
+    hostname: foo.com
+    protocol: HTTP
+    port: 80
+---
+apiVersion: gateway.networking.k8s.io/v1alpha1
+kind: ListenerSet
+metadata:
+  name: first-workload-listeners
+  labels:
+    gateway.networking.k8s.io/gateway.name: parent-gateway
+spec:
+  parentRef:
+    name: parent-gateway
+    kind: Gateway
+    group: gateway.networking.k8s.io
+  listeners:
+  - name: first
+    hostname: first.foo.com
+    protocol: HTTPS
+    port: 443
+    tls:
+      mode: Terminate
+      certificateRefs:
+      - kind: Secret
+        group: ""
+        name: first-workload-cert # Provisioned via HTTP01 challenge
+---
+apiVersion: gateway.networking.k8s.io/v1alpha1
+kind: ListenerSet
+metadata:
+  name: second-workload-listeners
+spec:
+  parentRef:
+    name: parent-gateway
```

Comment:
Are we sure we don't want this to be a list? I know it explodes the complexity of status to allow multiple parents here, but it's also something we need to be _very_ sure we can live without if we don't go with a list here.

Reply:
I'm good to make it a list - it's makes the resource similar to a Route.

I had some reservations earlier since I thought about ListenerSet inheriting policy from the Gateway (see: https://github.com/kubernetes-sigs/gateway-api/pull/3213#discussion_r1697172768).

Reply:
Though this does make it harder to lookup ListenerSets since I'll have to drop the labelling lookup

Reply:
made this a list